Content/Items/Gear/Gear.cs

```diff
@@ -217,4 +222,35 @@ public override void LoadData(TagCompound tag)
 			}
 		}
 	}
+	
+	/// <summary>
+	/// Used to apply on hit effects for affixes that have them
+	/// </summary>
+	/// <param name="player"></param>
+	/// <param name="target"></param>
+	/// <param name="hit"></param>
+	/// <param name="damageDone"></param>
+	public override void OnHitNPC(Player player, NPC target, NPC.HitInfo hit, int damageDone)

```

Comment:
Was thinking we could add an OnHitNPC override to UniversalBuffingPlayer; add "OnFireChance" as a field to "StatModifier", then apply it using the UniversalBuffingPlayer.
(so just increase the "OnFireChance" in the ChanceToApplyOnFireGearAffix class's apply affix)

This would let it stack better, and also show when hovering gear (as in the gear difference).

Reply:
I gave this a shot, but seeing the numbers increment constantly without being reset. Not sure how they get the correct ones. Need your assist on where that happens

Reply:
All set